python/cudf/cudf/core/column/column.py

```diff
@@ -391,41 +351,22 @@ def to_array(self, fillna=None):
         if ``fillna`` is ``None``, null values are skipped.  Therefore, the
         output size could be smaller.
         """
-        return self.to_dense_buffer(fillna=fillna).to_array()
+        return self.to_gpu_array(fillna=fillna).copy_to_host()
 
     @property
     def valid_count(self):
         """Number of non-null values"""
-        return len(self) - self._null_count
-
-    @property
-    def null_count(self):
-        """Number of null values"""
-        return self._null_count
-
-    @property
-    def has_null_mask(self):
-        """A boolean indicating whether a null-mask is needed"""
-        return self._mask is not None
+        return len(self) - self.null_count
 
     @property
     def nullmask(self):
         """The gpu buffer for the null-mask
         """
-        if self.has_null_mask:
-            return self._mask
+        if self.mask:
+            return cudf.Series(self._mask_view())
```

Comment:
Why are we returning a `cudf.Series` here now?

Reply:
Done